go/arrow/array/binarybuilder_test.go

```diff
@@ -85,3 +85,32 @@ func TestBinaryBuilder_ReserveData(t *testing.T) {
 	assert.Zero(t, ab.Cap(), "unexpected ArrayBuilder.Cap(), NewBinaryArray did not reset state")
 	assert.Zero(t, ab.NullN(), "unexpected ArrayBuilder.NullN(), NewBinaryArray did not reset state")
 }
+
+func TestBinaryBuilderReuse(t *testing.T) {
+	var (
+		mem = memory.NewGoAllocator()
```

Comment:
```suggestion
		mem = memory.NewCheckedAllocator(memory.NewGoAllocator())
```

and add the corresponding `defer` call:
```go
defer mem.AssertSize(t, 0)
```